src/context/WalletProvider/NativeWallet/NativePasswordRequired/hooks/useInitializeWalletFromStorage/useInitializeWalletFromStorage.ts

```diff
@@ -0,0 +1,25 @@
+import { NativeAdapter } from '@shapeshiftoss/hdwallet-native'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { useLocalStorage } from 'hooks/useLocalStorage/useLocalStorage'
+import { useEffect } from 'react'
+
+type StoredWallets = Record<string, string>
+
+export const useInitializeWalletFromStorage = () => {
+  const { state } = useWallet()
+  const [localStorageWallet] = useLocalStorage<StoredWallets>('wallet', {})
+
+  useEffect(() => {
+    if (!(localStorageWallet && state.adapters?.native)) return
+    ;(async () => {
+      for (const [deviceId] of Object.entries(localStorageWallet)) {
+        try {
+          const device = await (state.adapters?.native as NativeAdapter).pairDevice(deviceId)
+          await device?.initialize()
+        } catch (e) {
+          console.error('Error pairing native wallet', deviceId)
+        }
+      }
+    })()
+  }, [localStorageWallet, state.adapters])
```

Comment:
would this be better in `useWallet`? I guess this is very native wallet specific so maybe it's good here.

Reply:
This is only used in the Native view so I think that is why it is built the way that it is. And this code is apart of when a user reloads and is pulling the wallet from local storage. In this case their would not be a wallet in useWallet yet.